compression.go

```diff
@@ -0,0 +1,71 @@
+package kafka
+
+const (
+	CompressionNone int8 = iota
+	CompressionGZIP
+	CompressionSnappy
+	CompressionLZ4
+)
+
+var codecs map[int8]CompressionCodec
```

Comment:
It would be save to synchronize access to this map with a `sync.RWMutex` because we can't predict when a program will call `RegisterCompressionCodec`.